ThirdpartyToolchain.cmake

```diff
@@ -125,8 +125,11 @@ macro(build_protobuf)
 
     check_cxx_compiler_flag("-Winvalid-noreturn"
                             COMPILER_HAS_W_INVALID_NORETURN)
+
     if(COMPILER_HAS_W_INVALID_NORETURN)
       string(APPEND CMAKE_CXX_FLAGS " -Wno-invalid-noreturn")
+    else()
+      string(APPEND CMAKE_CXX_FLAGS " -Wno-error")
```

Comment:
Can you add a comment on which compiler/OS this happens on ?

Reply:
Will add, just for reference I've only reproduced with Ubuntu 22.04 and clang 14:
```
-- Building Protobuf from source
-- Performing Test COMPILER_HAS_W_INVALID_NORETURN
-- Performing Test COMPILER_HAS_W_INVALID_NORETURN - Failed
```